src/messaging/destinations/window-iframe-destination.js

```diff
@@ -0,0 +1,123 @@
+     *
+     * @type {string}
+     * @private
+     */
+    this._targetIframeID = targetIframeID
+
+    window.addEventListener('message', this._responseHandler.bind(this), false)
+  }
+
+  /**
+   * Registers a function to be called an a receiving side when a message from origin to destination will arrive.
+   * This callback will receive the following arguments: the request object (of `RequestMessage` type)
+   * and the function that can be used to send a response.
+   *
+   * @param {Function} callbackFn - A function that will be called when a request will arrive to its destination.
+   */
+  registerReceiverCallback (callbackFn) {
+    window.addEventListener('message', this._requestHandler.bind(this, callbackFn), false)
+  }
+
+  /**
+   * A function that will be called to send a request from origin to destination.
+   *
+   * @param {RequestMessage} requestMessage - A request message object.
+   */
+  sendRequest (requestMessage) {
+    const iframe = document.querySelector(`#${this._targetIframeID}`)
+    if (!iframe) {
+      throw new Error(`An #${this._targetIframeID} iframe does not exist in the document`)
+    }
+    const iframeWindow = iframe.contentWindow
+    iframeWindow.postMessage(requestMessage, this._targetURL)
+  }
+
+  /**
+   * A function that is used to send a response from destination to origin.
+   *
+   * @param {ResponseMessage} responseMessage - A response message object.
+   */
+  sendResponse (responseMessage) {
+    window.parent.postMessage(responseMessage, responseMessage.requestHeader.origin)
+  }
+
+  /**
+   * An internal handler that is called when request arrives to its destination.
+   *
+   * @param {Function} callbackFn - A client's callback function that will be called and
+   *                                passed a request (a `RequestMessage` object).
+   * @param {Event} event - A browser's event object.
+   * @private
+   */
+  _requestHandler (callbackFn, event) {
+    // `data` prop of an event contains a request message object
+    let request = event.data // eslint-disable-line prefer-const
+    request.header.origin = event.origin
+    callbackFn(request, this.sendResponse.bind(this))
+  }
+
+  /**
+   * An internal handler that is called when response arrives from destination to origin.
+   *
+   * @param {Event} event - A browser's event object.
+   * @private
+   */
+  _responseHandler (event) {
+    if (event.origin !== this._targetURL) {
+      // Message came from a destination we're not listening for
+      return
+    }
+
+    // `data` prop of an event contains a response message object
+    const responseMessage = event.data
+    if (this._responseCallback) {
+      this._responseCallback(responseMessage)
+    }
+  }
+}
+/*
+Below are some preset configurations for the most commonly used cases
+ */
+WindowIframeDestination.config = {}
+
+/*
+This configuration is used to access CEDICT data.
+ */
+WindowIframeDestination.config.CEDICT = {
```

Comment:
See comment in Cedict service. I believe this belongs there.

Reply:
Your understanding of how the whole thing works is absolutely correct. 

What you suggest about moving WindowIframeDestination config to Cedict Service makes total sense in a way that this is a data that characterizes Cedict service. On the other hand, I was thinking that one day a messaging service might by split from Cedict package into its own. That would be in line with our discussion about creating a generic messaging service. In that case, if all destination configs will be in the messaging service, one has to import just the messaging service and get all destinations available. That's why I was thinking about putting destination configs into the messaging service files.

However, we do not have such separation now. A messaging service is bundled with Cedict service and that whole package has to be imported anyway. So we could move config to Cedict service object for now and move it out and into the messaging service once we decide to split the latter.

What do you think? Does it make sense to split messaging service into a separate package in the future?

Reply:
I think it does make sense to think about splitting the messaging service into a separate package in the future, but even so, I'm not sure I would agree that even in that case we would want the configs to be bundled with the generic messaging service rather than the specific services. I could see an argument either way.  We don't necessarily have to decide this now, either, but I would like us to be thinking about the best way to make this functionality generic.  The most immediate next usage I see for it is to retrieve data from the Arethusa service running inside the diagram IFrame (see the following ongoing discussion on that topic for reference: https://docs.google.com/document/d/16YnkgSezIh-Sj1JMYpHghzXl3wTdbcYBR_beDmsqa_c/edit?usp=sharing)

Reply:
Agree with that. I moved the configuration object to the CEDICT service